src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/adminofficer/AdminOfficerPersonalisationProvider.java

```diff
@@ -31,6 +33,25 @@ public ImmutableMap<String, String> getDefaultPersonalisation(AsylumCase asylumC
             .build();
     }
 
+    public ImmutableMap<String, String> getAdminPersonalisation(AsylumCase asylumCase) {
+
+        final HearingCentre hearingCentre = asylumCase.read(AsylumCaseDefinition.HEARING_CENTRE, HearingCentre.class)
+                .orElseThrow(() -> new IllegalStateException("hearingCentre is not present"));
+        final AppealDecision appealOutcomeDecision = asylumCase
+                .read(AsylumCaseDefinition.IS_DECISION_ALLOWED, AppealDecision.class)
+                .orElseThrow(() -> new IllegalStateException("appealOutcomeDecision is not present"));
+
+        return ImmutableMap
+                .<String, String>builder()
+                .put("appellantGivenNames", asylumCase.read(AsylumCaseDefinition.APPELLANT_GIVEN_NAMES, String.class).orElse(""))
+                .put("appellantFamilyName", asylumCase.read(AsylumCaseDefinition.APPELLANT_FAMILY_NAME, String.class).orElse(""))
+                .put("appealReferenceNumber", asylumCase.read(AsylumCaseDefinition.APPEAL_REFERENCE_NUMBER, String.class).orElse(""))
+                .put("ariaListingReference", asylumCase.read(AsylumCaseDefinition.ARIA_LISTING_REFERENCE, String.class).orElse(""))
+                .put("hearingCentre", String.valueOf(hearingCentre).toUpperCase())
+                .put("applicationDecision", String.valueOf(appealOutcomeDecision).toUpperCase())
+                .build();
+    }
+
```

Comment:
This class will behave as a common service that can be called by multiple personalisations. Will all callers require hearingCentre and applicationDecision? If not, this will lead to a bad request exception from govNotify.

Common service should only provide common fields like appellant name or appeal ref number. Personalisation specific fields like hearingCentre or applicationDecision should be populated within individual personalisations.

Reply:
There's only one usage of this method at the moment, so this risk hopefully is negligible.
Furthermore, nothing stops the user of this method from overriding the ImmutableMap and use their own.
Regardless: the code will be modified to populate those two fields only when present, so that callers are not required to have those two properties available but, if they do, they will be populated. Any other refactoring need will be addressed "as needed".

Marking as resolved.